### ISSUE TITLE
Overfitting on custom dataset

### ISSUE THREAD

**Issue by hgaiser** [rbgirshick/py-faster-rcnn#156]
We are trying to learn on our own dataset which we recorded in a controlled environment, but the training algorithm quickly appears to overfit on the data. We recorded 5 objects with nearly 600 images per object, resulting in around 3000 images. If we compare the score on a test set after 500 iterations with that of 1000, 1500, 2000, etc. the result only seems to decrease. 

It appears the network is overfitting the training data, what is the suggested approach for this?

If we train on 500 iterations the network has only seen 500 images if I'm correct (IMS_PER_BATCH=1), so I am worried it will not be correctly trained because it might not have seen all of the training data.

EDIT: Test results on 500 iterations aren't bad by the way, but my worry is that it hasn't seen all of the training data.

EDIT 2: Results also seem rather unstable, as in between different runs (same number of iterations), MAP varies sometimes by 10%..

**Comment by catsdogone** [comment]
I think you iterations is fairly small. The result may be better if you that it runs several epochs.

**Comment by hgaiser** [comment]
Running with more iterations only makes overfitting worse. I believe my network was training on the black background instead of on the object. If I replace the black background with random images from pascal VOC I get greatly improved results (and I don't see overfitting, it appears to converge to some optimal network). So the lack of variability appeared to have been the issue.

**Comment by SaiAdityaG** [comment]
@hgaiser  I am facing the same issue while training on 10 classes(subset of VOC 2007). How did you check the background images.

**Comment by hgaiser** [comment]
I think you misunderstood what I men at. We made our own dataset by placing the items in a completely black environment. I believe the network was overtraining on this somehow. We segmented the black parts out of the image and replaced them with random images from VOC. If you train on a subset of VOC I doubt you'd have the same issue.

**Comment by ant1pink** [comment]
@hgaiser how did you solved your overfitting?
